pkgs/build-support/fetchgithub/default.nix

```
@@ -1,97 +1,165 @@
+      rev == null
+    )) "fetchFromGitHub requires one of either `rev` or `tag` to be provided (not both)."
+  );
 
-let
+  let
 
-  position = (if args.meta.description or null != null
-    then builtins.unsafeGetAttrPos "description" args.meta
-    else if tag != null then
-      builtins.unsafeGetAttrPos "tag" args
-    else
-      builtins.unsafeGetAttrPos "rev" args
-  );
-  baseUrl = "https://${githubBase}/${owner}/${repo}";
-  newMeta = meta // {
-    homepage = meta.homepage or baseUrl;
-  } // lib.optionalAttrs (position != null) {
-    # to indicate where derivation originates, similar to make-derivation.nix's mkDerivation
-    position = "${position.file}:${toString position.line}";
-  };
-  passthruAttrs = removeAttrs args [ "owner" "repo" "tag" "rev" "fetchSubmodules" "forceFetchGit" "private" "githubBase" "varPrefix" ];
-  varBase = "NIX${lib.optionalString (varPrefix != null) "_${varPrefix}"}_GITHUB_PRIVATE_";
-  useFetchGit = fetchSubmodules || (leaveDotGit == true) || deepClone || forceFetchGit || fetchLFS || (sparseCheckout != []);
-  # We prefer fetchzip in cases we don't need submodules as the hash
-  # is more stable in that case.
-  fetcher =
-    if useFetchGit then fetchgit
-    # fetchzip may not be overridable when using external tools, for example nix-prefetch
-    else if fetchzip ? override then fetchzip.override { withUnzip = false; }
-    else fetchzip;
-  privateAttrs = lib.optionalAttrs private {
-    netrcPhase =
-      let machineName = if githubBase == "github.com" then "api.github.com" else githubBase;
-      in ''
-        if [ -z "''$${varBase}USERNAME" -o -z "''$${varBase}PASSWORD" ]; then
-          echo "Error: Private fetchFromGitHub requires the nix building process (nix-daemon in multi user mode) to have the ${varBase}USERNAME and ${varBase}PASSWORD env vars set." >&2
-          exit 1
-        fi
-        cat > netrc <<EOF
-        machine ${machineName}
-                login ''$${varBase}USERNAME
-                password ''$${varBase}PASSWORD
-        EOF
-      '';
-    netrcImpureEnvVars = [ "${varBase}USERNAME" "${varBase}PASSWORD" ];
-  };
+    position = (
+      if args.meta.description or null != null then
+        builtins.unsafeGetAttrPos "description" args.meta
+      else if tag != null then
+        builtins.unsafeGetAttrPos "tag" args
+      else
+        builtins.unsafeGetAttrPos "rev" args
+    );
+    baseUrl = "https://${githubBase}/${owner}/${repo}";
+    newMeta =
+      meta
+      // {
+        homepage = meta.homepage or baseUrl;
+      }
+      // lib.optionalAttrs (position != null) {
+        # to indicate where derivation originates, similar to make-derivation.nix's mkDerivation
+        position = "${position.file}:${toString position.line}";
+      }
+      // lib.optionalAttrs (tag != null) {
+        # GitHub exposes tag/release descriptions suitable for referencing
+        changelog = meta.changelog or "${baseUrl}/releases/tag/${tag}/CHANGELOG.md";
```

```suggestion
        changelog = meta.changelog or "${baseUrl}/releases/tag/${tag}/CHANGELOG.md" or null;
```

This enables `meta = { ...; inherit (finalAttrs.src) changelog; ... };` which does not clobber `unsafeGetAttrPos` while still allowing `src` to be overridden to use `rev`

---

(in this case it isn't a `optionalAttrs`, just add it always)

---

Fixed, thanks for the improvement and sorry for the delay.